parsec/parsec.c

```diff
@@ -2584,10 +2560,11 @@ int parsec_check_overlapping_binding(parsec_context_t *context)
 static int parsec_parse_comm_binding_parameter(const char* option, parsec_context_t* context)
 {
 #if defined(PARSEC_HAVE_HWLOC)
-    if( option[0]!='\0' ) {
+    if( option[0] != '\0' ) {
         int core = atoi(option);
-        if( (core > -1) && (core < parsec_hwloc_nb_real_cores()) )
-            context->comm_th_core = core;
+        /* negative core allowed to force an absolute core selection */
```

Comment:
This is not true, passing a negative value to `find_core_by_idx` will cause an infinite loop.